packages/remote-feature-flag-controller/src/remote-feature-flag-controller.test.ts

```diff
@@ -640,11 +645,154 @@ describe('RemoteFeatureFlagController', () => {
+        const controller = createController();
+
+        controller.setFlagOverride('testFlag', true);
+
+        expect(controller.state.localOverrides).toStrictEqual({
+          testFlag: true,
+        });
+      });
+
+      it('overwrites existing override for the same flag', () => {
+        const controller = createController();
+
+        controller.setFlagOverride('testFlag', true);
+        controller.setFlagOverride('testFlag', false);
+
+        expect(controller.state.localOverrides).toStrictEqual({
+          testFlag: false,
+        });
+      });
+
+      it('preserves other overrides when setting a new one', () => {
+        const controller = createController();
+
+        controller.setFlagOverride('flag1', 'value1');
+        controller.setFlagOverride('flag2', 'value2');
+
+        expect(controller.state.localOverrides).toStrictEqual({
+          flag1: 'value1',
+          flag2: 'value2',
+        });
+      });
+    });
+
+    describe('clearFlagOverride', () => {
+      it('removes a specific override', () => {
+        const controller = createController();
+
+        controller.setFlagOverride('flag1', 'value1');
+        controller.setFlagOverride('flag2', 'value2');
+        controller.clearFlagOverride('flag1');
+
+        expect(controller.state.localOverrides).toStrictEqual({
+          flag2: 'value2',
+        });
+      });
+
+      it('does not affect state when clearing non-existent override', () => {
+        const controller = createController();
+
+        controller.setFlagOverride('flag1', 'value1');
+        controller.clearFlagOverride('nonExistentFlag');
+
+        expect(controller.state.localOverrides).toStrictEqual({
+          flag1: 'value1',
+        });
+      });
+    });
+
+    describe('clearAllOverrides', () => {
+      it('removes all overrides', () => {
+        const controller = createController();
+
+        controller.setFlagOverride('flag1', 'value1');
+        controller.setFlagOverride('flag2', 'value2');
+        controller.clearAllOverrides();
+
+        expect(controller.state.localOverrides).toStrictEqual({});
+      });
+
+      it('does not affect state when no overrides exist', () => {
+        const controller = createController();
+
+        controller.clearAllOverrides();
+
+        expect(controller.state.localOverrides).toStrictEqual({});
+      });
+    });
+
+    describe('integration with remote flags', () => {
+      it('preserves overrides when remote flags are updated', async () => {
+        const clientConfigApiService = buildClientConfigApiService({
+          remoteFeatureFlags: { remoteFlag: 'initialRemoteValue' },
+        });
+        const controller = createController({ clientConfigApiService });
+
+        // Set override before fetching remote flags
+        controller.setFlagOverride('overrideFlag', 'overrideValue');
+
+        await controller.updateRemoteFeatureFlags();
+
+        // Mock different remote response for second fetch
```

Comment:
This mentions that we are mocking the request for the second time, but where are we mocking the request for the first time?